kotest-assertions/kotest-assertions-shared/src/jsHostedMain/kotlin/io/kotest/assertions/print/printers.kt

```diff
@@ -0,0 +1,5 @@
+package io.kotest.assertions.print
+
+// js doesn't support full reflection so we just need to delegate to the instances toString
+actual fun <A : Any> dataClassPrint(): Print<A> = ToStringPrint
+actual fun <A : Any> platformPrint(a: A): Print<A>? = null
```

Comment:
What is jsHosted?

Reply:
`jsHosted` is the common source set for Kotlin/JS and Kotlin/Wasm/JS, both of which operate within a JS host environment. In the near future, there will be Kotlin/Wasm/WASI (not JS-hosted).

Reply:
Defined here: https://github.com/kotest/kotest/pull/3805/files#diff-8eba8e3034dadebe1b9bae3190de6c5de838ea157d15ac6fe48d54291883bf0a

Reply:
By within a js host, does that mean executes within a JS engine/vm?
And Kotlin/Wasm/WASI will not require a js engine ?

Reply:
1. Yes, a JS host means that a JS engine is available, providing access to JS APIs (though the bridging process slows things down). The JS engine also shares its garbage collector with GC-based languages targeting Wasm (like Kotlin).
2. Yes. Pure WASI runtimes like [Wasmtime](https://github.com/bytecodealliance/wasmtime) or [WasmEdge](https://wasmedge.org/) have no JS engine at all. These seem to become the fast (microsecond startup times), lightweight, language-independent, universal virtualization runtimes of the future (no more Docker). (There's also Node.js, which does have a V8-based WASI runtime.)

Reply:
Ticket to track: [KT-60278](https://youtrack.jetbrains.com/issue/KT-60278) Make Kotlin/Wasm suitable for standalone Wasm VMs